masu/database/provider_status_accessor.py

```diff
@@ -0,0 +1,212 @@
+                            'status': ProviderStatusCode.READY,
+                            'last_message': 'none',
+                            'timestamp': datetime.now(timezone.utc),
+                            'retries': 0}
+            self.add(**ready_status)
+            self.commit()
+            self._obj = self._get_db_obj_query().first()
+
+    # pylint: disable=arguments-differ
+    def _get_db_obj_query(self):
+        """
+        Return the sqlachemy query for the provider object.
+
+        Args:
+            None
+        Returns:
+            (sqlalchemy.orm.query.Query): "SELECT public.api_customer.group_ptr_id ..."
+        """
+        provider_accessor = ProviderDBAccessor(self._provider_uuid)
+        self.provider = provider_accessor.get_provider()
+        if not self.provider:
+            raise MasuProviderError(f'Unknown provider: {self._provider_uuid}')
+        return super()._get_db_obj_query(provider_id=self.provider.id)
+
+    def get_status(self):
+        """
+        Return the provider status.
+
+        Args:
+            None
+        Returns:
+            (int): Status code
+        """
+        return self._obj.status
+
+    def get_provider_uuid(self):
+        """
+        Return the provider uuid.
+
+        Args:
+            None
+        Returns:
+            (uuid.uuid4): Provider uuid
+        """
+        return self._provider_uuid
+
+    def get_last_message(self):
+        """
+        Return the provider last_message.
+
+        Args:
+            None
+        Returns:
+            (str): Error message, if any
+        """
+        return self._obj.last_message
+
+    def get_timestamp(self):
+        """
+        Return the provider status timestamp.
+
+        Args:
+            None
+        Returns:
+            (datetime): Timestamp for the status
+        """
+        return self._obj.timestamp
+
+    def get_retries(self):
+        """
+        Return the provider status retry count.
+
+        Args:
+            None
+        Returns:
+            (int): Status retry count
+        """
+        return self._obj.retries
+
+    def is_backing_off(self):
+        """Determine if the provider is waiting to retry."""
+        def backoff(interval, maximum=64):
+            """Exponential back-off."""
+            # (12 + interval) is used to select desirable values from the
+            # exponential progression.
+            return min(maximum, (2 ** (12 + interval))) + (random.randint(0, 1000) / 1000.0)
+
+        retries = self.get_retries()
+        timestamp = self.get_timestamp()
+        backoff_threshold = timestamp + timedelta(seconds=backoff(retries,
+                                                                  maximum=86400))
+        if self.get_status() == ProviderStatusCode.WARNING and \
+                retries <= ProviderStatusAccessor.MAX_RETRIES and \
+                datetime.now(timezone.utc) <= backoff_threshold:
+            return True
```

Comment:
Need log statements for each state transition but especially when we back off from attempting an ingest for a provider

Reply:
Why? What's the concern?